compat/getaddrinfo.cc

```diff
@@ -105,18 +105,15 @@ xgetaddrinfo (const char *nodename, const char *servname,
     struct servent *servent;
     const char *socktype;
     int port;
-    struct addrinfo hint, result;
-    struct addrinfo *ai, *sai, *eai;
+    struct addrinfo hint = {} , result = {};
+    struct addrinfo *ai = nullptr, *sai = nullptr, *eai = nullptr;
     char **addrs;
```

Comment:
I do not object to these improvements, but I think that if we modernize this code, we should do more:

* one complete type/name/value declaration statement per line (no cost here and often reduces future diffs)
* move the variable declaration as close to its use as possible/reasonable instead of preserving legacy hoisting to the top of the function (often clarifies code a lot and sometimes prevents bugs)
* cover all variables in the function (initializing `sai` while leaving, say, `servent` pointer uninitialized makes the reader waste time on wondering whether the inconsistency is meaningful)

Reply:
To avoid misunderstanding: You have the right to limit the scope of this PR to memset() elimination, of course. However, in that case, you should not touch variable declarations at all -- just replace memset(x) with `x = {}` assignments. I do _not_ recommend doing that, but it would at least present a consistent, minimal scope unlike the current strange mix of assorted and seemingly unfinished improvements.

Reply:
Fair enough. Let me modernize all the things; back soon with an update